httpcore/__init__.py

```diff
@@ -1,7 +1,7 @@
-from ._async.base import AsyncByteStream, AsyncHTTPTransport
+from ._async.base import AsyncHTTPTransport
```

Comment:
I guess we just need to be careful that this drops `SyncByteStream` and `AsyncByteStream`, which we rely on in a few places in HTTPX. Some other implementations out there would also still be relying on these interfaces.

Should we perhaps keep some aliases in place until 1.0…?

```python
from typing import Iterable, AsyncIterable

SyncByteStream = Iterable[bytes]
AsyncByteStream = AsyncIterable[bytes]
```

Reply:
Yeah I don't think we're sufficiently confident yet about exactly where we're going with this interface,
so I reckon rolling the other stuff (where we *are* confident) and keeping this hanging a little longer.